server/map_rpc_server.go

```diff
@@ -56,62 +56,42 @@ func (t *TrillianMapServer) IsHealthy() error {
 }
 
 // Init creates the initial revision 0 SignedMapHead, if one doesn't already exist.
-func (t *TrillianMapServer) Init(ctx context.Context, mapID int64) error {
+func (t *TrillianMapServer) Init(ctx context.Context, mapID int64) (*trillian.SignedMapRoot, error) {
 	tree, hasher, err := t.getTreeAndHasher(ctx, mapID, false /* readonly */)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	ctx = trees.NewContext(ctx, tree)
 
 	tx, err := t.registry.MapStorage.BeginForTree(ctx, mapID)
 	if err != storage.ErrMapNeedsInit && err != nil {
-		return err
+		return nil, err
 	}
 	defer tx.Close()
 
 	if err == nil {
 		// Init() not needed.
-		return nil
+		return nil, nil
```

Comment:
When `Init` is called _after_ the log has already been initialized, should it return the SMR of revision 0 rather than nil? 

A description of how clients would use the SMR might help determine what a consistent response from `Init` should be.

Reply:
I don't think so, no - this way you can tell if whether or not the Init() did anything.
I'd expect that when a fresh SMR is created, the operator using the `CreateTree` tool might want to inspect/verify it - if the Map's already been initialised, then there's no need to do that again (or they can use the GetSignedMapRootAtRevision() to inspect it if they have some other reason).

Reply:
If there's a use case for doing something special then this makes sense.  

It intuitively feels like it doesn't match typical REST API or singleton init semantics which always return a valid thing when called, or error when called twice. I'd worry a bit that callers would try to dereference the response if `err==nil` and result in a crash.

Reply:
Good call - `InitMap()` now returns an AlreadyExists error if the map was already initialised.